onnxruntime/core/providers/webnn/builders/helper.h

```diff
@@ -39,46 +41,76 @@ std::string GetShapeString(std::vector<T>& shape) {
   return shape_info.str();
 }
 
+template <typename T>
+bool ReadIntArrayFrom1DTensor(const onnx::TensorProto& tensor, std::vector<T>& array, const logging::Logger& logger) {
+  std::vector<uint8_t> unpacked_tensor;
+  auto status = onnxruntime::utils::UnpackInitializerData(tensor, unpacked_tensor);
+  if (!status.IsOK()) {
+    LOGS(logger, ERROR) << "Error while unpacking shape: " << status.ErrorMessage();
+    return false;
+  }
+  const auto& dims = tensor.dims();
+  if (dims.size() != 1) {
+    LOGS(logger, VERBOSE) << "The tensor must be 1D.";
+    return false;
+  }
+  int64_t rank = dims[0];
+  if (tensor.data_type() != ONNXTensorElementDataType::ONNX_TENSOR_ELEMENT_DATA_TYPE_INT64) {
+    LOGS(logger, VERBOSE) << "The type of tensor's element data must be INT64.";
+    return false;
+  }
+  const int64_t* array_data = reinterpret_cast<const int64_t*>(unpacked_tensor.data());
+  if constexpr (std::is_same<T, int64_t>::value) {
+    array.assign(array_data, array_data + rank);
+  } else if constexpr (std::is_same<T, int32_t>::value) {
+    std::transform(array_data, array_data + rank,
+                   std::back_inserter(array),
+                   [](int64_t dim) -> int32_t { return SafeInt<int32_t>(dim); });
+  }
+  return true;
+}
+
 bool IsInputSupported(const NodeArg& node_arg, const std::string& parent_name, const logging::Logger& logger);
 
 // Get a list of groups of supported nodes, each group represents a subgraph supported by WebNN EP.
 std::vector<std::vector<NodeIndex>> GetSupportedNodes(const GraphViewer& graph_viewer,
                                                       const emscripten::val& wnn_builder_,
                                                       const logging::Logger& logger);
 static const InlinedHashMap<std::string, std::string> op_map = {
-    {"ArgMax", "argMax"},
-    {"ArgMin", "argMin"},
-    {"Add", "add"},
-    {"Sub", "sub"},
-    {"Mul", "mul"},
-    {"Div", "div"},
-    {"Pow", "pow"},
-    {"Cos", "cos"},
-    {"Erf", "erf"},
-    {"Floor", "floor"},
-    {"Sin", "sin"},
-    {"Sqrt", "sqrt"},
-    {"Relu", "relu"},
-    {"LeakyRelu", "leakyRelu"},
-    {"Sigmoid", "sigmoid"},
-    {"Softmax", "softmax"},
-    {"Cast", "cast"},
-    {"Clip", "clamp"},
-    {"Conv", "conv2d"},
-    {"ConvTranspose", "convTranspose2d"},
-    {"Concat", "concat"},
+    {"ArgMax", "argMax1"},
+    {"ArgMin", "argMin1"},
```

Comment:
Oops, I merged too fast, please remove these debug code in a follow-up. Thanks!

Reply:
Fixed with new PR https://github.com/Honry/onnxruntime/pull/8, PTAL, thanks.

Reply:
Ooh, I wondered about these, but didn't realize it was debug code.

Reply:
That's a tip use for fallbacking ops to CPU EP.